warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -319,51 +337,70 @@ private Set<Excerpt> getExcerpts(PhraseIndexes phraseIndexes) {
+        int i = 0;
+        while (i <= 1) {
+            // set the start of the range
+            if (i == 0) {
+                excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(start));
+            } else {
+                if (start - expandSize > 0) {
+                    excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(start - expandSize));
                 } else {
-                    log.warn(TermFrequencyExcerptIterator.class.getSimpleName() + " returned top key with incorrectly-formatted column qualifier in key: "
-                                    + topKey + " when scanning for excerpt [" + start + "," + end + "] for field " + field + " within range " + range);
-                    return null;
+                    excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(0));
                 }
-            } else {
-                return null;
             }
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to scan for excerpt [" + start + "," + end + "] for field " + field + " within range " + range, e);
-        }
-    }
-
-    private String getHitPhrase(ArrayList<String> hitTermValues, String[] phraseParts) {
-        List<String> hitPhrase = new ArrayList<>();
-        for (String phrasePart : phraseParts[1].split(Constants.SPACE)) {
-            if (hitTermValues.contains(phrasePart)) {
-                hitPhrase.add("[" + phrasePart + "]");
+            if (i == 0) {
+                excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end));
             } else {
-                hitPhrase.add(phrasePart);
+                // set the end of the range
+                excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end + expandSize));
             }
-        }
+            if (log.isDebugEnabled() && i != 0) {
+                log.debug("size of excerpt requested: {}", excerptFields.getOffset(field) * 2);
+                log.debug("original range is ({},{}) and the expanded range is ({},{})", start, end,
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.START_OFFSET),
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.END_OFFSET));
+            }
+            try {
+                excerptIterator.init(source, excerptIteratorOptions, env);
+                excerptIterator.setHitTermsList(hitTermValues);
+                excerptIterator.setDirection(excerptFields.getDirection(field).toUpperCase().trim());
+                excerptIterator.setOrigHalfSize(origHalfSize);
+                if (i != 0) {
+                    excerptIterator.setTrim(true);
+                }
+                excerptIterator.seek(range, Collections.emptyList(), false);
+                // if an excerpt is returned...
+                if (excerptIterator.hasTop()) {
```

Comment:
If I'm reading this section correctly, it looks like we'll never return an excerpt if the attempt is == 0 - we alwasy seem to return null in that case. Is this correct behavior? Aren't there some cases where we'll want to return what is produced by the excerptIterator during the first attempt?

Reply:
it will return an excerpt during attempt==0 if there are no stop words found when creating wordsAndScoresArr (the list of all of the terms we can choose from at all of the offsets in range)